app/src/main/java/com/android/feedme/model/viewmodel/CameraViewModel.kt

```diff
@@ -185,6 +214,59 @@ class CameraViewModel : ViewModel() {
         }
       }
 
+  /**
+   * Performs object labelling on the given bitmap.
+   *
+   * This function processes the given bitmap to detect objects and extracts labels from the
+   * detected objects. If the labelling is successful, it updates the ingredient list and returns
+   * the best label. If an error occurs or no labels are found, it updates the error message
+   * accordingly.
+   *
+   * @param bitmap The bitmap of the photo to label.
+   * @return A string representing the best labelling result, or null if labelling fails or no
+   *   labels are found.
+   */
+  private suspend fun performObjectLabelling(bitmap: Bitmap): String? {
+    return withTimeoutOrNull(10000L) {
+      suspendCancellableCoroutine { continuation ->
+        val labelList = mutableMapOf<String, Float>()
+        var errorObjectOccurred = false
+
+        objectExtraction(
+            bitmap = bitmap,
+            onSuccess = { detectedObjects ->
+              labelList.putAll(labelProcessing(detectedObjects))
+              if (labelList.isNotEmpty()) {
+                val label = bestLabel(labelList)
+                if (label.isNotEmpty()) {
+                  updateIngredientList(
+                      IngredientMetaData(
+                          0.0, MeasureUnit.NONE, Ingredient(label, "NO_ID", false, false)))
+                  continuation.resume(label)
+                } else {
+                  _errorToDisplay.value = ERROR_NO_FOOD_LABEL
+                  onAnalyzeDone()
+                  continuation.resume(null)
+                }
+              } else {
+                _errorToDisplay.value = ERROR_NO_LABEL
+                onAnalyzeDone()
+                continuation.resume(null)
+              }
+            },
+            onFailure = {
+              _errorToDisplay.value = ERROR_NO_OBJECT
+              errorObjectOccurred = true
+              continuation.resume(null)
+            })
+      }
+    }
+        ?: run {
+          _errorToDisplay.value = ERR_TIMEOUT
```

Comment:
I think this is the line of the error bug. You can start by looking here

Reply:
Ok I'm fixing that now ! Thank you :)